### DISCUSSION TITLE
Currency display in Widget: EUR -> €

### DISCUSSION THREAD

**Discussion by alain0** [pretix/pretix#4649] [Q&A]
Is there is a way to display the currency with a symbol instead of the text abbreviation in the widget; so € instead of EUR 
(e.g. https://docs.pretix.eu/en/latest/user/events/widget.html#example).

On the shop pages there is in fact a € sign.

**Comment by wiffbi** [comment]
I am sorry, currently no. The widget has a hard-coded `currency + " " + price` format. I think we should default to what the shop does – although that is a breaking change :( I guess it would be nice to add html-structure to the price output as well, so it could be further formatted through CSS if one wants to.

Instead of introducing a breaking change, we could add a basic format-string to be provided through an attribute, e.g. `<pretix-widget money-format="{} €">…`. That wouldn’t allow for a stylable HTML-structure, but that wouldn’t be needed, as one could just use `money-format="{}"` and use CSS’s `.pretix-widget-pricebox>span:after` selector with `content: "€"` to add a stylable currency element. 🤔

**Reply by raphaelm** [reply]
> I am sorry, currently no. The widget has a hard-coded `currency + " " + price` format. I think we should default to what the shop does – although that is a breaking change :( 

The problem so far was not that it's a breaking change, but that we can't easily fix it. In the shop, we render it with python-babel, which gives us a format for every combination of currency and locale. For example, Australian Dollar is rendered as $ in an Australian event and AU$ in a US event. Euro is rendered 12,00 € in German but €12.00 in English. For the Widget, we'd need to implement the entire set of possibilities manually in JavaScript, which did not seem worth it to me

**Reply by wiffbi** [reply]
You are right, that is definitely overkill. I thought of handing over a money-format in product_list – but extracting the format from python-babel is probably not so easy, espcially since it also affects number_format 🤔  … Would a `money-format` attribute still be feasible? Or is that too much of a foot-gun for organizers?

**Reply by raphaelm** [reply]
It feels a little too specific to me to do this but not 100 other things we could allow to customize in the widget

**Reply by alain0** [reply]
Thanks for the replies!
I would be happy with this very specific attr, but I understand you don't want to open the door for 1000 more of these tiny settings..
Just a span around the currency and price would already give me enough options to customise things.

**Reply by wiffbi** [reply]
I will look into adding a span around currency and price each. I am currently working on the widget anyways to improve accessibility …

**Comment by wiffbi** [comment]
We [added](https://github.com/pretix/pretix/commit/e5f29bd59203489fbd976c47ee6447ea28af16f0) a `span.pretix-widget-pricebox-currency` around the currency, so you can style/hide it to your liking. I hope that helps.

**Reply by alain0** [reply]
Great, thank you!
